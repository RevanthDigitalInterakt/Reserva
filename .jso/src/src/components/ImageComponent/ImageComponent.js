  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _objectDestructuringEmpty2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[2]));
  var _reactNativeAutoHeightImage = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[3]));
  var _configDeviceSizes = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function ImageComponent(_ref) {
    var props = Object.assign({}, ((0, _objectDestructuringEmpty2.default)(_ref), _ref));
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[5]).jsx)(_reactNativeAutoHeightImage.default, Object.assign({
      resizeMode: "contain",
      width: _configDeviceSizes.default.DEVICE_WIDTH
    }, props));
  }
  var _default = exports.default = (0, _react.memo)(ImageComponent);
