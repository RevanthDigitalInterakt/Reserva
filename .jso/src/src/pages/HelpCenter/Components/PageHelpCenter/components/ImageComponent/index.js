  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = ImageComponent;
  var _react = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[2]);
  var _configDeviceSizes = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[3]));
  var _styles = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  function ImageComponent(_ref) {
    var _data$items$, _images$items;
    var data = _ref.data;
    if (!(data != null && data.items)) return null;
    var images = data == null ? undefined : (_data$items$ = data.items[0]) == null ? undefined : _data$items$.helpCenterImagesCollection;
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[5]).jsx)(_reactNative.View, {
      children: images && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[5]).jsx)(_reactNative.View, {
        style: _styles.default.containerImage,
        children: images == null ? undefined : (_images$items = images.items) == null ? undefined : _images$items.map(function (image) {
          return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[5]).jsx)(_reactNative.Image, {
            source: {
              uri: image == null ? undefined : image.url
            },
            style: {
              height: _configDeviceSizes.default.DEVICE_WIDTH * 0.5
            },
            resizeMode: "contain"
          }, `image-helpCenter-${image == null ? undefined : image.url}`);
        })
      })
    });
  }
