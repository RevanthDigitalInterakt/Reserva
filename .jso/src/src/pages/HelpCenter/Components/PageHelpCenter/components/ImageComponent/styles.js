  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _reactNative = _$$_REQUIRE(_dependencyMap[0]);
  var styles = _reactNative.StyleSheet.create({
    containerImage: {
      marginBottom: 8
    }
  });
  var _default = exports.default = styles;
