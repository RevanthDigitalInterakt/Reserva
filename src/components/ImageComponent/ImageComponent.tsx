import React, { memo } from 'react';
import AutoHeightImage from 'react-native-auto-height-image';
import configDeviceSizes from '../../utils/configDeviceSizes';

export interface TSource {
  uri: string;
}

interface IProps extends Omit<React.ComponentProps<typeof AutoHeightImage>, 'source' | 'width'> {
  source: TSource | number
  width?: number;
}

function ImageComponent({ ...props }: IProps) {
  return (
    <AutoHeightImage
      resizeMode="contain"
      width={configDeviceSizes.DEVICE_WIDTH}
      {...props}
    />
  );
}

export default memo(ImageComponent);
