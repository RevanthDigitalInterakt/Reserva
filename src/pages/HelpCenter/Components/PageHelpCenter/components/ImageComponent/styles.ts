import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  containerImage: {
    marginBottom: 8,
  },
});

export default styles;
