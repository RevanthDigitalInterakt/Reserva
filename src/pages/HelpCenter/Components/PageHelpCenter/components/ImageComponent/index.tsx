import React from 'react';
import { Image, View } from 'react-native';
import configDeviceSizes from '../../../../../../utils/configDeviceSizes';
import styles from './styles';
import type { BodyImagesCollectionOutput, Maybe } from '../../../../../../base/graphql/generated';

interface IImageProps {
  data: Maybe<BodyImagesCollectionOutput>;
}

export default function ImageComponent({ data }: IImageProps) {
  if (!data?.items) return null;

  const images = data?.items[0]?.helpCenterImagesCollection;

  return (
    <View>
      {images && (
        <View
          style={styles.containerImage}
        >
          {images?.items?.map((image) => (
            <Image
              key={`image-helpCenter-${image?.url}`}
              source={{ uri: image?.url }}
              style={{
                height: configDeviceSizes.DEVICE_WIDTH * 0.5,
              }}
              resizeMode="contain"
            />
          ))}
        </View>
      )}
    </View>
  );
}
