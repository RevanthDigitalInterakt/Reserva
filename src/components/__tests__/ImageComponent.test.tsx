/* eslint-disable global-require */
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import ImageComponent from '../ImageComponent/ImageComponent';

describe('ImageComponent', () => {
  it('should match with the snapshot', async () => {
    const Component = (
      <ImageComponent
        source={require('../../../assets/common/banner-login.png')}
      />
    );

    const { toJSON } = await waitFor(() => render(Component));
    expect(toJSON()).toMatchSnapshot();
  });
});
