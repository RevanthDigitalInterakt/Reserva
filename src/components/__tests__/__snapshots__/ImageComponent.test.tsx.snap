// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent should match with the snapshot 1`] = `
<Image
  onError={[Function]}
  resizeMode="contain"
  source={
    {
      "commons": {
        "GuideBermuda": [Circular],
        "GuideCalca": [Circular],
        "GuideCamiseta": [Circular],
        "GuideHoodie": [Circular],
        "GuideJaqueta": [Circular],
        "GuideMangaCurta": [Circular],
        "GuideMangaLonga": [Circular],
        "GuidePolo": [Circular],
        "GuideShort": [Circular],
        "GuideSueter": [Circular],
        "GuideSungaCueca": [Circular],
        "QrcodeBackground": [Circular],
        "abandonedCartBag": [Circular],
        "addition": [Circular],
        "americanExpress": [Circular],
        "arrowInstruction": [Circular],
        "arrowRight": [Circular],
        "bagEmpty": [Circular],
        "bannerCatalog": [Circular],
        "bannerLogin": [Circular],
        "bannerOffer": [Circular],
        "black1p5p": [Circular],
        "cardFront": [Circular],
        "cardImage": [Circular],
        "cardVerso": [Circular],
        "chat": [Circular],
        "checkedBox": [Circular],
        "chevronLeftFill": [Circular],
        "chevronRightFill": [Circular],
        "close": [Circular],
        "clothingCare1": [Circular],
        "clothingCareBleach": [Circular],
        "clothingCareDrying": [Circular],
        "clothingCarePass": [Circular],
        "clothingCareProfessionalWashing": [Circular],
        "clothingCareWash": [Circular],
        "config": [Circular],
        "copy": [Circular],
        "cupomModalBackground": [Circular],
        "cutlery": [Circular],
        "discountFlag": [Circular],
        "dividerReserva": [Circular],
        "document": [Circular],
        "elo": [Circular],
        "emptyProductCatalog": [Circular],
        "expand": [Circular],
        "facebook": [Circular],
        "faq": [Circular],
        "filledHeart": [Circular],
        "foraDoAr": [Circular],
        "green1p5p": [Circular],
        "greenCheck": [Circular],
        "headerLogin": [Circular],
        "heart": [Circular],
        "help": [Circular],
        "homeMock1": [Circular],
        "homeMock2": [Circular],
        "homeMock3": [Circular],
        "homeMock4": [Circular],
        "imageCashback": [Circular],
        "imageNotFound": [Circular],
        "instagram": [Circular],
        "ios14": [Circular],
        "localReserva": [Circular],
        "logo": [Circular],
        "logoQRCode": [Circular],
        "mastercard": [Circular],
        "newsletter": [Circular],
        "noStoresFound": [Circular],
        "noWishList": [Circular],
        "pinPlace": [Circular],
        "pinYou": [Circular],
        "profile": [Circular],
        "purchaseConfirmation": [Circular],
        "raceImageBackground": [Circular],
        "reservaIcon": [Circular],
        "roulet": [Circular],
        "saleOff": [Circular],
        "search": [Circular],
        "searchNotFound": [Circular],
        "selectRectangle": [Circular],
        "share": [Circular],
        "shipping1": [Circular],
        "shipping2": [Circular],
        "shirt1": [Circular],
        "shirt2": [Circular],
        "shirt3": [Circular],
        "shirt4": [Circular],
        "signup1": [Circular],
        "signup2": [Circular],
        "signup3": [Circular],
        "signup4": [Circular],
        "signup5": [Circular],
        "subtraction": [Circular],
        "topBannerMock1": [Circular],
        "twitter": [Circular],
        "umPCincoPLogo": [Circular],
        "uncheckedBox": [Circular],
        "unfilledHeart": [Circular],
        "update": [Circular],
        "visa": [Circular],
        "whatsapp": [Circular],
        "withoutInternet": [Circular],
        "zoomHand": [Circular],
      },
      "default": {
        "GuideBermuda": [Circular],
        "GuideCalca": [Circular],
        "GuideCamiseta": [Circular],
        "GuideHoodie": [Circular],
        "GuideJaqueta": [Circular],
        "GuideMangaCurta": [Circular],
        "GuideMangaLonga": [Circular],
        "GuidePolo": [Circular],
        "GuideShort": [Circular],
        "GuideSueter": [Circular],
        "GuideSungaCueca": [Circular],
        "QrcodeBackground": [Circular],
        "abandonedCartBag": [Circular],
        "addition": [Circular],
        "americanExpress": [Circular],
        "arrowInstruction": [Circular],
        "arrowRight": [Circular],
        "bagEmpty": [Circular],
        "bannerCatalog": [Circular],
        "bannerLogin": [Circular],
        "bannerOffer": [Circular],
        "black1p5p": [Circular],
        "cardFront": [Circular],
        "cardImage": [Circular],
        "cardVerso": [Circular],
        "chat": [Circular],
        "checkedBox": [Circular],
        "checkedRounded": [Function],
        "chevronLeftFill": [Circular],
        "chevronRight": [Function],
        "chevronRightFill": [Circular],
        "close": [Circular],
        "closeIcon": [Function],
        "clothingCare1": [Circular],
        "clothingCareBleach": [Circular],
        "clothingCareDrying": [Circular],
        "clothingCarePass": [Circular],
        "clothingCareProfessionalWashing": [Circular],
        "clothingCareWash": [Circular],
        "config": [Circular],
        "copy": [Circular],
        "cupomModalBackground": [Circular],
        "cutlery": [Circular],
        "discountFlag": [Circular],
        "dividerReserva": [Circular],
        "document": [Circular],
        "elo": [Circular],
        "emptyProductCatalog": [Circular],
        "expand": [Circular],
        "facebook": [Circular],
        "faq": [Circular],
        "filledHeart": [Circular],
        "foraDoAr": [Circular],
        "green1p5p": [Circular],
        "greenCheck": [Circular],
        "hanger": [Function],
        "headerLogin": [Circular],
        "heart": [Circular],
        "help": [Circular],
        "homeMock1": [Circular],
        "homeMock2": [Circular],
        "homeMock3": [Circular],
        "homeMock4": [Circular],
        "imageCashback": [Circular],
        "imageNotFound": [Circular],
        "info": [Function],
        "infoSuccess": [Function],
        "instagram": [Circular],
        "ios14": [Circular],
        "localReserva": [Circular],
        "logo": [Circular],
        "logoPrime": [Function],
        "logoQRCode": [Circular],
        "mastercard": [Circular],
        "newsletter": [Circular],
        "noStoresFound": [Circular],
        "noWishList": [Circular],
        "personalize": [Function],
        "pinPlace": [Circular],
        "pinYou": [Circular],
        "profile": [Circular],
        "purchaseConfirmation": [Circular],
        "raceImageBackground": [Circular],
        "reservaIcon": [Circular],
        "roulet": [Circular],
        "saleOff": [Circular],
        "search": [Circular],
        "searchNotFound": [Circular],
        "selectRectangle": [Circular],
        "share": [Circular],
        "shipping1": [Circular],
        "shipping2": [Circular],
        "shirt1": [Circular],
        "shirt2": [Circular],
        "shirt3": [Circular],
        "shirt4": [Circular],
        "signup1": [Circular],
        "signup2": [Circular],
        "signup3": [Circular],
        "signup4": [Circular],
        "signup5": [Circular],
        "sizeGuideIcon": [Function],
        "subtraction": [Circular],
        "topBannerMock1": [Circular],
        "trash": [Function],
        "twitter": [Circular],
        "umPCincoPLogo": [Circular],
        "uncheckedBox": [Circular],
        "unfilledHeart": [Circular],
        "update": [Circular],
        "visa": [Circular],
        "whatsapp": [Circular],
        "withoutInternet": [Circular],
        "zoomHand": [Circular],
      },
      "svgs": {
        "checkedRounded": [Function],
        "chevronRight": [Function],
        "closeIcon": [Function],
        "hanger": [Function],
        "info": [Function],
        "infoSuccess": [Function],
        "logoPrime": [Function],
        "personalize": [Function],
        "sizeGuideIcon": [Function],
        "trash": [Function],
      },
    }
  }
  style={
    [
      {
        "height": 562.5,
        "width": 750,
      },
      undefined,
    ]
  }
/>
`;
